UPGRADE-1.12.md

```diff
@@ -32,6 +32,12 @@ and should be used only this way.
 are now included in every env starting with `test` keyword. If you wish to not have them, either you need to rename your env to not start 
 with test or remove these services with complier pass.
 
+6. The following listeners have been removed as they are not used anymore:
+- `Sylius\Bundle\CoreBundle\CatalogPromotion\Listener\CatalogPromotionFailedListener`
+- `Sylius\Bundle\CoreBundle\CatalogPromotion\Listener\CatalogPromotionUpdateFailedMessageListener`
```

Comment:
I'm wondering if we should remove it 🤔 In theory, listeners are not included in our [BC policy](https://docs.sylius.com/en/1.12/book/organization/backward-compatibility-promise.html?highlight=BC%20policy#event-listeners), but their result is (`Behaviour they’re providing (the end result) is still included in BC promise.`). Therefore, it's possible one could already implement functionality in their application that uses these listeners and the removed event as well. It's obviously a dead code in Sylius, but does not necessarily have to be such in the project that is using Sylius.
On the other hand, it would be sad if we have to keep these unused classes until Sylius 2.0 in our codebase, so it's tempting to remove them :D I need a second opinion cc @lchrusciel @GSadee

Reply:
I would be for removing them as listeners are excluded from the BC policy (as you wrote) and these ones do not provide any behaviour for the Sylius application without customization as they were a dead code. However, I'm not sure if it is a proper approach 🤔

Reply:
Just thinking about it, I would vote for removal, as well. Plus - if someone needs it, duplication is super straightforward, we can even add it to UPGRADE file